File: app/src/main/java/com/mmartin/ghibliapi/film/detail/FilmDetailContract.java
package com.mmartin.ghibliapi.film.detail;

import com.mmartin.ghibliapi.BasePresenter;
import com.mmartin.ghibliapi.BaseView;

/**
 * Created by mmartin on 8/12/17.
 */

public interface FilmDetailContract {
    interface View extends BaseView {
        void showMissingFilm();
        void showTitle(String title);
        void showDescription(String description);
        void showDirector(String director);
        void showProducer(String producer);
        void showPeople(); //List of string/People??
    }

    interface Presenter extends BasePresenter<View> {
        void setFilmId(String filmId);
    }
}

File: app/src/main/java/com/mmartin/ghibliapi/film/detail/FilmDetailContract.kt
package com.mmartin.ghibliapi.film.detail

import com.mmartin.ghibliapi.BasePresenter
import com.mmartin.ghibliapi.BaseView

/**
 * Contract interfaces for film detail presenter and view
 *
 * Created by mmartin on 8/12/17.
 */

interface FilmDetailContract {
    interface View : BaseView {
        fun showMissingFilm()
        fun showTitle(title: String)
        fun showDescription(description: String)
        fun showDirector(director: String)
        fun showProducer(producer: String)
        fun showPeople(people: List<Pair<String, String>>)
    }

    interface Presenter : BasePresenter<View>
}
